### ISSUE TITLE
Accessibility issues

### ISSUE THREAD

**Issue by shrilakshmishastry** [react-tags/react-tags#999]
Hey @ad1992 👋  Hope you're doing good. Here are the accessibility issues I have noticed in react-tags . 

- tags are not reachable to edit using the keyboard
- suggestions are not screen reader accessible, when you move across the suggestions VO doesn’t announce it
- Clear all buttons doesn’t explain what to clear. it feels like I am trying to clear texts in input but actually, it does is clear tags
- screen reader should announce the suggestions when the suggestion is shown
- should provide flexibility to add a label to input tag
- proper aria label for tags and nested interaction handle
- when tags are read only do we need to make tags keyboard accessible? most probably not, when they are read-only they will behave as normal text. This should be considered when fixing keyboard accessibility in tags

I want to pick all of these and fix them. Can you please assign me this issue?

**Comment by ad1992** [comment]
@shrilakshmishastry really appreciate your effort in listing the issues. Please go ahead and fix it ❤️
As per readonly tags, yes they should be just keyboard accessible as they are regular text when readonly

**Comment by shrilakshmishastry** [comment]
Hey @ad1992 what do you mean by keyboard accessible? 

tags shouldn't be a tab/focusable element right when read-only?

**Comment by ad1992** [comment]
> Hey @ad1992 what do you mean by keyboard accessible?
> 
> tags shouldn't be a tab/focusable element right when read-only?

Nope, in readonly its just rendering the tags in `div` so individual tags won't be focussible, however it should be announced via SR

**Comment by shrilakshmishastry** [comment]
Hey @ad1992 👋  hope you're doing well. I need your opinion on this 

- nested interaction in tags

The tag as a whole is keyboard-reachable and activates when clicking enter. On active it behaves as an input tag. When active close icon will be hidden. 

The close icon is also in the tab index sequence when the whole tag acts as a clickable. 

I guess the tag should be a button rather than a span.

**Comment by shrilakshmishastry** [comment]
Hey @ad1992 , here is the solution for the above issues that I am planning to implement. Can you please give me a correction?

Solutions:

- Clear all buttons don’t explain what to clear. it feels like I am trying to clear texts in input but actually, it does clear tags

Give a prop to provide an alt tag to clear all CTA in case CTA is provided. If not give the default alt tag as “clear all chosen tags”

- should provide flexibility to add labels to input tag
    
    This is the label for the main input tag. Users don’t know what they are inputting for. Should make label compulsory. 
    
- proper aria label for tags

This is in the context where tags are read-only. if VO users go through this they don’t get the context of tags. Get the reference/ID to label for tags as props. It’s optional. If a label is not provided what will you do? in this case, it makes sense to make the label compulsory 

- nested interaction in tags

The tag as a whole is keyboard-reachable and activates when clicking enter. On active it behaves as an input tag. When active close icon will be hidden. 

The close icon is in the next number in the tab index when the whole tag acts as a clickable. 

I guess it should be a button rather span. 

- suggestions are not screen reader accessible, when you move across the suggestions VO doesn’t announce it

This is happening as we are moving across the non-activable element. Usually, VO when reading across the text of the website using ctrl+options+shift  , reads the content in any tag. But for tab indexable elements if we are using non-indexable elements, then we need to take care of it. Giving an aria-label to suggestions fixes this.  The suggestion itself should be used as aria label.  

- screen reader should announce the count of suggestions when the suggestion is shown

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role

Need to follow the above pattern and aria-live should be added to the suggestion container. So when it’s open it can announce the count of suggestions 

- when tags are readonly do we need to make the tags keyboard accessible? most probably not, when they are readonly they will behave as normal text. This should be considered when fixing keyboard accessibility in tags

Yes, when tags are read-only they should behave like any text on screen. 

how shall we handle it as both tag and read-only is handled in a single tag?
This can be handled by adding tab-index=0 when ready so the tags are not tab-reachable and any style applied for activation will not be applied to it.
